openupgrade_scripts/scripts/website_sale_slides/16.0.1.0/pre-migration.py

```diff
@@ -0,0 +1,23 @@
+# Copyright 2025 Tecnativa - Carlos Lopez
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from openupgradelib import openupgrade
+
+
+def _update_detailed_type_to_course(env):
+    """
+    Adapt all products using into a course to the new detailed_type=course
+    """
+    openupgrade.logged_query(
+        env.cr,
+        """UPDATE product_template pt
+            SET detailed_type = 'course'
+        FROM product_product pp
+            JOIN slide_channel sc ON sc.product_id = pp.id
+        WHERE pp.product_tmpl_id = pt.id;
```

Comment:
```suggestion
        WHERE pp.product_tmpl_id = pt.id AND pt.type = 'service';
```
not sure if this is needed...

Reply:
The field doesn't have a specific domain, so you can select any product. Therefore, I think the additional WHERE clause is not necessary.

![image](https://github.com/user-attachments/assets/106cf994-902f-4439-b710-020ae5cf38ff)

![image](https://github.com/user-attachments/assets/380407aa-b1bd-4803-b798-2c5ea0c5a381)

Reply:
Interesting.